src/Core/Batch/BatchRunner.php

```diff
@@ -0,0 +1,183 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Core\Batch;
+
+/**
+ * A class for executing jobs in batch.
+ */
+class BatchRunner
+{
+    use SysvTrait;
+
+    /**
+     * @var BatchConfig
+     */
+    private $config;
+
+    /**
+     * @var ConfigStorageInterface
+     */
+    private $configStorage;
+
+    /**
+     * @var JobSubmitInterface
+     */
+    private $submitter;
+
+    /**
+     * Determine internal implementation and loads the configuration.
+     *
+     * @param ConfigStorageInterface $configStorage [optional] The
+     *        ConfigStorage object to use. **Defaults to** null.
+     * @param JobSubmitInterface $jobSubmitter [optional] The JobSubmitter
+     *        object to use. **Defaults to** null.
+     */
+    public function __construct(
+        ConfigStorageInterface $configStorage = null,
+        JobSubmitInterface $jobSubmitter = null
+    ) {
+        if ($configStorage === null || $jobSubmitter === null) {
+            if ($this->isSysvIPCLoaded() && $this->isDaemonRunning()) {
+                $configStorage = new SysvConfigStorage();
+                $jobSubmitter = new SysvJobSubmitter();
+            } else {
+                $configStorage = InMemoryConfigStorage::getInstance();
+                $jobSubmitter = $configStorage;
+            }
+        }
+        $this->configStorage = $configStorage;
+        $this->submitter = $jobSubmitter;
+        $this->loadConfig();
+    }
+
+    /**
+     * Register a job for batch execution.
+     *
+     * @param string $identifier Unique identifier of the job.
+     * @param callable $func Any Callable except for Closure. The callable
+     *        should accept an array of items as the first argument.
+     * @param array $options [optional] {
+     *     Configuration options.
+     *
+     *     @type int $batchSize The size of the batch.
+     *     @type int $callPeriod The period in seconds from the last execution
+     *               to force executing the job.
+     *     @type int $workerNum The number of child processes. It only takes
+     *               effect with the {@see \Google\Cloud\Core\BatchDaemon}.
+     *     @type string $bootstrapFile A file to load before executing the
+     *                  job. It's needed for registering global functions.
+     * }
+     * @return boolean true on success, false on failure
+     */
+    public function registerJob($identifier, $func, $options = [])
+    {
+        if ($func instanceof \Closure) {
+            trigger_error('Closure is not allowed', E_USER_NOTICE);
```

Comment:
I'm not sure that a E_USER_NOTICE is high enough. Lots of PHP configurations don't display notices by default (ugh), so an invalid configuration could silently fail. Perhaps E_USER_ERROR or a \InvalidArgumentException would be better.